src/clj/game/cards/programs.clj

```diff
@@ -877,4 +877,48 @@
                    :counter-cost [:power 3]
                    :once :per-turn
                    :msg "gain [Click][Click]"
-                   :effect (effect (gain :click 2))}]}})
+                   :effect (effect (gain :click 2))}]}
+
+   "Wari"
+   {:events {:successful-run
+             {:interactive (req )
+              :delayed-completion true
+              :req (req (and (= target :hq)
+                             (first-successful-run-on-server? state :hq)
+                             (some #(and (ice? %)
+                                         (not (rezzed? %)))
+                                   (all-installed state :corp))))
+              :effect (effect
+                       (continue-ability
+                        {:prompt "Use Wari?"
+                         :choices ["Yes" "No"]
+                         :delayed-completion true
+                         :effect (effect 
+                                  (continue-ability
+                                   (if (= target "Yes")
+                                     {:prompt "Choose a subtype"
+                                      :choices ["Barrier" "Code Gate" "Sentry"]
+                                      :delayed-completion true
+                                      :effect (effect
+                                               (trash card)
```

Comment:
This needs to be `{:unpreventable true}`, as it's a "trash __to__ do this" wording. To handle chain effects from the trashing (say... Q-Coherence Chip, causing trash-prevention from SacCon), you will want to await `(trash state side card {:unpreventable true})` with `when-completed`, the continue the ability. Getting tricky now!